packages/paste-core/components/anchor/src/index.tsx

```diff
@@ -1,48 +1,53 @@
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
 import {StyledLink} from './styles';
-import {AnchorProps} from './types';
+
+export type AnchorTargets = '_self' | '_blank' | '_parent' | '_top';
+export type AnchorTabIndexes = 0 | -1;
+
+interface Anchor {
+  className?: never;
+  children: NonNullable<React.ReactNode>;
+  href: string;
+  id?: never;
+  onBlur?(event: React.FocusEvent<HTMLElement>): void;
+  onClick?(event: React.MouseEvent<HTMLElement>): void;
+  onFocus?(event: React.FocusEvent<HTMLElement>): void;
+  rel?: string;
+  tabIndex?: AnchorTabIndexes;
+  target?: AnchorTargets;
+}
 
 const EXTERNAL_LINK_REGEX = /^(https?:)[^\s]*$/;
 const EXTERNAL_TARGET_DEFAULT = '_blank';
 const EXTERNAL_REL_DEFAULT = 'noreferrer noopener';
 
 const isExternalUrl = (url: string): boolean => EXTERNAL_LINK_REGEX.test(url);
 
-const handlePropValidation = ({href, tabIndex, children}: AnchorProps): void => {
-  const hasHref = href != null && href !== '';
-  const hasTabIndex = tabIndex != null;
-
-  if (!hasHref) {
-    throw new Error(
-      `[Paste: Anchor] Missing href prop for anchor. Maybe you're looking for the [Paste: Button] component.`
-    );
-  }
-
-  if (children == null) {
-    throw new Error(`[Paste: Anchor] Must have non-null children.`);
-  }
-
-  if (hasTabIndex && !(tabIndex === 0 || tabIndex === -1)) {
-    throw new Error(`[Paste: Anchor] TabIndex must be 0 or -1.`);
-  }
+const Anchor: React.FC<Anchor> = props => (
+  <StyledLink
+    href={props.href}
+    rel={isExternalUrl(props.href) && !props.rel ? EXTERNAL_REL_DEFAULT : props.rel}
+    onBlur={props.onBlur}
+    onClick={props.onClick}
+    onFocus={props.onFocus}
+    tabIndex={props.tabIndex}
+    target={isExternalUrl(props.href) && !props.target ? EXTERNAL_TARGET_DEFAULT : props.target}
+  >
+    {props.children}
+  </StyledLink>
+);
+
+Anchor.propTypes = {
+  children: PropTypes.node.isRequired,
+  tabIndex: PropTypes.oneOf([0, -1]),
+  target: PropTypes.oneOf(['_self', '_blank', '_parent', '_top']),
 };
 
-const Anchor: React.FC<AnchorProps> = props => {
-  handlePropValidation(props);
-
-  return (
-    <StyledLink
-      href={props.href}
-      rel={isExternalUrl(props.href) && !props.rel ? EXTERNAL_REL_DEFAULT : props.rel}
-      onBlur={props.onBlur}
-      onClick={props.onClick}
-      onFocus={props.onFocus}
-      tabIndex={props.tabIndex}
-      target={isExternalUrl(props.href) && !props.target ? EXTERNAL_TARGET_DEFAULT : props.target}
-    >
-      {props.children}
-    </StyledLink>
-  );
+Anchor.defaultProps = {
```

Comment:
As these are optional I'm not sure we need to declare default values for them, given they are the browser defaults. 

Thoughts?

Reply:
@SiTaggart if i remove the default props, I get an lint error: https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/require-default-props.md

Reply:
huh, that lint rule isn't very sympathetic to HTML attributes. 

Leave it for now, just seems weird to apply `target` and `tabindex` to an anchor when there's really no need to.

Reply:
I'd be into just removing the rule, this situation brings it's usefulness (or lack of) to light.

Reply:
Turning off the rule.